### PR TITLE
fix: removed hca specific config from lungmap (#4287)

### LINKED ISSUE #4287: [LungMAP DX] Remove HCA specific config from the LungMAP Explorer
### Need
LungMAP currently has some config that it is inheriting from HCA that should not be there.

#### Entity List
1. Biological Network, filter and column
2. Access filter

image

#### Project Detail
1. Atlas

image

### Approach
* Create LM-specific `projectsEntityConfig`, `samplesEntityConfig`, and `filesEntityConfig`, removing HCA-specific values defined above.
* Create LM-specific `CATEGORY_GROUP`, removing HCA-specific values defined above.
* For both columns and filters, reuse definitions/const as much as possible but exclude HCA-specific values. 
* Follow directory structure, naming conventions etc of HCA config.
* Remove HCA-specific config as described above.

### PR DESCRIPTION
**Pull request by mrtopsyt**
### Ticket

Closes #4287 .

### Reviewers

@frano-m .

### Changes

- Excluded the "Access" and "Biological Network" HCA filters from appearing on Lungmap
- Excluded the "Biological Network" HCA column from appearing on the Lungmap Project Entity List
- Excluded the "Atlas" HCA detail from appearing on the Lungmap Project Detail page
- Duplicated code from HCA for details page columns, index columns, and filters, to avoiud changes to HCA also affecting Lungmap

### CODE REVIEW

File: site-config/lungmap/dev/index/common/category.ts
Comment (frano-m) [inline site-config/lungmap/dev/index/common/category.ts:1]: Niiiice thank you @jpaten !

File: site-config/lungmap/dev/index/projectsEntityConfig.ts [hunk 1]
```diff
@@ -1,28 +1,47 @@
...
+} from "site-config/hca-dcp/category";
+import { ProjectsResponse } from "../../../../app/apis/azul/hca-dcp/common/responses";
+import {
+  getProjectId,
+  getTitle,
+} from "../../../../app/apis/azul/hca-dcp/common/utils";
+import * as C from "../../../../app/components";
+import { projectEdits as hcaProjectEdits } from "../../../../app/viewModelBuilders/azul/hca-dcp/common/projectMapper/projectEdits/projectEdits";
+import * as V from "../../../../app/viewModelBuilders/azul/hca-dcp/common/viewModelBuilders";
 import { mainColumn as hcaMetadataMainColumn } from "../../../hca-dcp/dev/detail/project/metadataMainColumn";
```
Comment (frano-m) [inline site-config/lungmap/dev/index/projectsEntityConfig.ts]: Hi @jpaten is there is a reason why we haven't copied across `metadataMainColumn` like we have for `metadataSideColumn`?

File: site-config/lungmap/dev/index/samplesEntityConfig.ts [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+      },
+    ],
+    defaultSort: {
+      desc: SORT_DIRECTION.ASCENDING,
+      id: HCA_DCP_CATEGORY_KEY.SAMPLE_ID,
+    },
+  } as ListConfig<SamplesResponse>,
+  listView: {
+    subTitleHero: hcaSubTitleHero,
+  },
```
Comment (frano-m) [inline site-config/lungmap/dev/index/samplesEntityConfig.ts]: Hi @jpaten, would you mind removing the `listView` from `samplesEntityConfig` lines `207-209`.

File: site-config/lungmap/dev/index/projectsEntityConfig.ts [hunk 3]
```diff
@@ -48,5 +67,197 @@ export const projectsEntityConfig: EntityConfig = {
...
+      },
+    ],
+    defaultSort: {
+      desc: SORT_DIRECTION.DESCENDING,
+      id: HCA_DCP_CATEGORY_KEY.AGGREGATE_LAST_MODIFIED_DATE,
+    },
+  },
+  listView: {
+    subTitleHero: hcaSubTitleHero,
+  },
```
Comment (frano-m) [inline site-config/lungmap/dev/index/projectsEntityConfig.ts]: Hi @jpaten would you mind removing `listView` from the `projectsEntityConfig`; lines `258-260`.

File: site-config/lungmap/dev/index/filesEntityConfig.ts [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+        id: HCA_DCP_CATEGORY_KEY.CELL_COUNT,
+        width: { max: "1fr", min: "124px" },
+      },
+    ],
+    defaultSort: {
+      desc: SORT_DIRECTION.ASCENDING,
+      id: HCA_DCP_CATEGORY_KEY.FILE_NAME,
+    },
+  } as ListConfig<FilesResponse>,
+  listView: {
```
Comment (frano-m) [inline site-config/lungmap/dev/index/filesEntityConfig.ts]: Hi @jpaten would you mind removing `listView` from `filesEntityConfig`. Thank you!

**Review by frano-m — APPROVED** [review]
Hi @jpaten nice work! I have a few change requests - removal of bits that LungMAP shouldn't inherit from HCA.

**Review by frano-m — CHANGES_REQUESTED** [review]
Hi sorry @jpaten - just a few minor changes and this PR should be good to go!

File: site-config/lungmap/dev/detail/project/mainColumn.ts [hunk 5]
```diff
@@ -2,30 +2,79 @@ import {
...
+    children: [
+      {
+        component: MDX.DataReleasePolicy,
+      } as ComponentConfig<typeof MDX.DataReleasePolicy>,
+    ],
+    component: C.CollapsableSection,
+    props: {
+      collapsable: false,
+      title: "Data Access Policy",
+    },
```
Comment (MillenniumFalconMechanic) [inline site-config/lungmap/dev/detail/project/mainColumn.ts]: It looks like the `Data Access Policy` config is duped here:

Screenshot 2024-12-02 at 11 14 31 AM

**Review by MillenniumFalconMechanic — CHANGES_REQUESTED** [review]
Hi @jpaten, this is looking good - thank you! I have one change request and then I think it should be good to go.

File: site-config/lungmap/dev/detail/project/mainColumn.ts (hunk 5 above)
Comment (mrtopsyt) [reply site-config/lungmap/dev/detail/project/mainColumn.ts]: @MillenniumFalconMechanic whoops - should be good now! Thanks!

**Review by frano-m — APPROVED** [review]
👏🏽 WOOT thank you @jpaten !

**Review by frano-m — APPROVED** [review]

**Review by MillenniumFalconMechanic — APPROVED** [review]
Thanks!

**Review by MillenniumFalconMechanic — COMMENTED** [review]
LGTM, thanks Jonah!

**Pull request merged**